### PR TITLE
mixRelease: improve the implementation

### PR DESCRIPTION
**Pull request by c4710n**
## Description of changes

This PR includes lots of changes for improving mixRelease.

> Currently, I'm implementing them as separated commits, which is helpful to review by others. But, if necessary, I can squash them into one single commit.

Every commit is self-explanatory. You can get more details by reading the description of commits.

Highlights:

+ investigate the reason why erlang is always referrenced.
+ add new option - `removeCookie` (default to `true`, which is for backward compatibility).
+ wrap programs in $out/bin with their runtime deps.
+ format the code and comments in a more consistent way.

Feel free to comment, and I can improve this PR as needed.

> @happysalada , I found that you are the primary maintainer of this file, so I @ you. Hope that I'm not disturbing you.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/beam-modules/mix-release.nix [hunk 1]
```diff
@@ -10,80 +25,104 @@
...
 
-  # mix dependencies generated by mix2nix
-  # this assumes each dependency is built by buildMix or buildRebar3
-  # each dependency needs to have a setup hook to add the lib path to $ERL_LIBS
-  # this is how mix will find dependencies
+  # Mix dependencies generated by mix2nix
+  #
+  # This assumes each dependency is built by buildMix or buildRebar3. Each
+  # dependency needs to have a setup hook to add the lib path to $ERL_LIBS.
+  # This is how Mix find dependencies.
```
Comment (happysalada) [inline pkgs/development/beam-modules/mix-release.nix:35]:
```suggestion
  # This is how Mix finds dependencies.
```

**Comment by happysalada** [comment]
The general cleanup looks good to me.
Just one comment, be careful when formatting a file. Right now there is an RFC in nixpkgs to decide exactly of the formatting, and the formatter still hasn't been chosen.
Usually formatting commits are not accepted. (that RFC should finish at some point, but it could easily take another 6 months - 1 year.
As to the commits, they are very detailed which is really nice for the review. If you don't mind though, I'll squash them upon merging.
I'm leaving this open for another 24 hours just in case more people want to have a look at it.

**Comment by c4710n** [comment]
Thank you for enlightening me about the background knowledge related to format.

And, I have committed what you suggest.

**Comment by happysalada** [comment]
forgot to tag @NixOS/beam 
I'm going to wait for 24h more just to give people who want time to review.

**Pull request merged**